### PR TITLE
suggesion marker

### PR DESCRIPTION
**Pull request by wanxxz**
show suggesion markers on map

### CODE REVIEW

File: src/api/search.ts [hunk 1]
```diff
@@ -52,28 +52,8 @@ export async function search(o: SearchParams, s: SearchServiceKey = 'openstreetm
...
-        source: 'tomtom',
-        data: v,
-        position: [v.position.lat, v.position.lon],
-        viewport: [
-          [v.viewport.btmRightPoint.lat, v.viewport.btmRightPoint.lon],
-          [v.viewport.topLeftPoint.lat, v.viewport.topLeftPoint.lon],
-        ],
-      }))
+    if (s === 'openstreetmap') return d
+    if (s === 'tomtom') return d.results.map(v => ({ display_name: v.address.freeformAddress, ...v.position }))
```
Comment (quinncnl) [inline src/api/search.ts:56]: Why revert this part? It is a mistake? We want to move to the viewport/boundingbox of search result.

Comment (wanxxz) [reply src/api/search.ts]: these returned value been used some where else besides suggestion ui/ux e.g. nearby/POI place

now using `zod` to parse return value and for type-inference [8f6b679](https://github.com/tracestrack/tracesmap.com/pull/7/commits/8f6b67972cec664b07c679d03bdde0a032c36740)

File: src/hooks/map.ts [hunk 2]
```diff
@@ -577,31 +584,16 @@ export function useControl() {
...
-    const neLL = fromLonLat([Number(p1.lon), Number(p1.lat)])
-    const swLL = fromLonLat([Number(p2.lon), Number(p2.lat)])
-    let points = new MultiPoint([neLL, swLL])
-
-    console.log(points)
-
-    view.fit(points)
-  }
-  return { zoomIn, zoomOut, setSearchResultPosition, setBoundingBox }
+  return { zoomIn, zoomOut, moveCenter }
```
Comment (quinncnl) [inline src/hooks/map.ts:596]: We should move to boundingbox/viewport and show selected search result here.

Comment (wanxxz) [reply src/hooks/map.ts:594]: https://github.com/tracestrack/tracesmap.com/pull/7/commits/b8acc422aa0486fa7cfbc3c6f76243cfbec86606

**Pull request merged**